activesupport/lib/active_support/file_evented_update_checker.rb

```diff
@@ -0,0 +1,72 @@
+require 'listen'
+
+module ActiveSupport
+  class FileEventedUpdateChecker
+    attr_reader :listener
+    def initialize(files, dirs={}, &block)
+      @files = Set.new
+      @files = files.map { |f| File.expand_path(f)}
+      @dirs = Hash.new
+      dirs.map do |key,value|
```

Comment:
The loop for `@files` with `map` builds a collection with the expanded paths, but the loop for `dirs` does not need to build anything. The return value of the `map` call is discarded. This tells you that iterating with `each` is enough.